adapters/src/main/java/io/realm/RealmBaseAdapter.java

```diff
@@ -61,6 +64,23 @@ public void onChange(BaseRealm results) {
         }
     }
 
+    public RealmBaseAdapter(@Nullable OrderedRealmCollection<T> data) {
+        if (data != null && !data.isManaged())
```

Comment:
I think we should also check if the data is managed in `RealmBaseAdapter(@NonNull Context context, @Nullable OrderedRealmCollection<T> data)` and `RealmRecyclerViewAdapter(@NonNull Context context, @Nullable OrderedRealmCollection<T> data, boolean autoUpdate)`.